src/misc/support.cpp

```diff
@@ -50,6 +50,14 @@ char int_to_char(int val)
 	return static_cast<char>(val);
 }
 
+uint8_t drive_index(char drive)
+{
+	const auto drive_letter = int_to_char(toupper(drive));
+	// Confirm the provided drive is valid
+	assert(drive_letter >= 'A' && drive_letter <= 'Z');
+	return static_cast<uint8_t>(drive_letter - 'A');
```

Comment:
Sounds great. 

Right now I'm using the assert as a canary to see if this really is the case; because all of the old code did  `val - 'A'`, and barreled on without any care what the result was.

But so far I can't trigger the assert to fail, even deliberately trying to use out-of-bounds drive values and lots of game testing- so this optimization is a good next-step!